lib/iris/plot.py

```diff
@@ -249,7 +249,19 @@ def _string_coord_axis_tick_labels(string_axes, axes=None):
 
     ax = axes if axes else plt.gca()
     for axis, ticks in string_axes.items():
-        formatter = mpl_ticker.IndexFormatter(ticks)
+        # Define a tick formatter, this replaces the now deprecated
+        # mpl_ticker.IndexFormatter. Note that this formatter behaves
+        # differently to IndexFormatter and will only assign labels to
```

Comment:
I don't think it's right to talk about a change in behaviour in a comment - that's only appropriate for the PR and the "What's New". Do you have some simple context on why `FuncFormatter` needs an input function like this?

Reply:
I've tried to clarify what the function is doing by making some of the variables more descriptive. I'm hoping that this + the new comment should make it clear how FuncFormatter is using this function. (i.e. that this function takes in the location of a tick and returns a label.) I felt that making it too explicit what was going on would have made the comment a little too verbose.

Reply:
That's super helpful. I had been seeing `FuncFormatter` as a closed magic box but I appreciate much better what's going on now.